lib/hg-utils.coffee

```diff
@@ -242,7 +242,11 @@ class Repository
       params = []
     if !util.isArray(params)
       params = [params]
-    child = spawnSync('hg', params)
+
+    return '' unless @isCommandForRepo(params)
+
+    options = { cwd: @rootPath }
+    child = spawnSync('hg', params, options)
```

Comment:
I can only assume if we return promises from `hgCommand` then we can probably get the same performance that the `GitRespositoryAsync` gets. Maybe create a new `hgCommandAsync` method specifically for the `HgRespositoryAsync` class?

I updated my hg a few days ago, so that should be fine. Think it might be a case of doing the platform check.